pkg/tarmak/cmd.go

```diff
@@ -103,6 +104,118 @@ func (c *CmdTarmak) Destroy() error {
 	return nil
 }
 
+func (c *CmdTarmak) DestroyEnvironment() error {
+	inputDestroy := input.New(os.Stdin, os.Stdout)
+	var err error
+	response := true
+
+	if !c.flags.Environment.Destroy.AutoApprove {
+		response, err = inputDestroy.AskYesNo(&input.AskYesNo{
+			Default: false,
+			Query:   fmt.Sprintf("Destroy all clusters of %v?", c.Environment().Name()),
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	if response {
+		c.log.Info("destroying clusters")
+		for _, cluster := range c.Environment().Clusters() {
+			// We first want to destroy the k8s clusters before the hub
+			if cluster.Name() != c.Environment().Hub().Name() {
+				c.cluster = cluster
+				c.log.Infof("destroying cluster %v", c.Cluster().Name())
+				if err := c.Destroy(); err != nil {
+					return err
+				}
+				if err = c.DestroyClusterMetadata(); err != nil {
+					return err
+				}
+			}
+		}
+
+		// After destroying all other clusters, we can destroy the hub
+		c.cluster = c.Environment().Hub()
+		c.log.Infof("destroying cluster %v", c.Cluster().Name())
+		if err := c.Destroy(); err != nil {
+			return err
+		}
+		if err = c.DestroyClusterMetadata(); err != nil {
+			return err
+		}
+	} else {
+		for _, cluster := range c.Environment().Clusters() {
+			hosts, err := cluster.ListHosts()
+			if err != nil {
+				return err
+			}
+			if len(hosts) > 0 {
+				return fmt.Errorf("can't proceed with destroying this environment, because it still has hosts running")
+			}
+		}
+	}
+
+	if !c.flags.Environment.Destroy.AutoApprove {
+		response, err = inputDestroy.AskYesNo(&input.AskYesNo{
+			Default: false,
+			Query:   "Move environment folder (SSH key and vault_root_token) to .archive?",
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	if response {
```

Comment:
The logic for these responses is not quite right. You need to at least set response to true again before checking auto-approve

Reply:
Yes sorry, you are quite right